lib/index.js

```diff
@@ -193,11 +200,11 @@ Socket.prototype._startPollSend = function () {
 }
 
 Socket.prototype._startPollReceive = function () {
-    if (!this._pollReceive) {
-        this._pollReceive = nn.PollReceiveSocket(this.binding, function (events) {
-            if (events) this._receive();
-        }.bind(this));
-    }
+  if (!this._pollReceive) {
+    this._pollReceive = nn.PollReceiveSocket(this.binding, function () {
+      this.push(this._receive());
+    }.bind(this));
```

Comment:
the descriptor ready check actually happens here, https://github.com/nickdesaulniers/node-nanomsg/blob/master/src/node_nanomsg.cc#L257 that's what lets the callback happen, anyway it calls NanomsgReadable and in there technically the callback fires.

either way the callback never fires unless it's readable. i'm pretty sure.